thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/utils/DatasetTriggerInfoRepo.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.pinot.thirdeye.datalayer.dto.DatasetConfigDTO;
+import org.apache.pinot.thirdeye.datalayer.dto.DetectionConfigDTO;
+import org.apache.pinot.thirdeye.datalayer.dto.MetricConfigDTO;
+import org.apache.pinot.thirdeye.datasource.DAORegistry;
+import org.apache.pinot.thirdeye.formatter.DetectionConfigFormatter;
+import org.apache.pinot.thirdeye.rootcause.impl.MetricEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DatasetTriggerInfoRepo {
+  private static final Logger LOG = LoggerFactory.getLogger(DatasetTriggerInfoRepo.class);
+  private static DatasetTriggerInfoRepo _instance = null;
+  private static final DAORegistry DAO_REGISTRY = DAORegistry.getInstance();
+  private Map<String, Long> datasetRefreshTimeMap;
+  private Set<String> dataSourceWhitelist;
+  private ScheduledThreadPoolExecutor executorService;
+
+  private DatasetTriggerInfoRepo(int refreshFreqMin, Collection<String> dataSourceWhitelist) {
+    this.datasetRefreshTimeMap = new ConcurrentHashMap<>();
+    this.dataSourceWhitelist = new HashSet<>(dataSourceWhitelist);
+    this.refresh(); // initial refresh
+    this.executorService = new ScheduledThreadPoolExecutor(1, r -> {
+      Thread t = Executors.defaultThreadFactory().newThread(r);
+      t.setDaemon(true);
+      return t;
+    });
+    this.executorService.scheduleAtFixedRate(
+        this::refresh, refreshFreqMin, refreshFreqMin, TimeUnit.MINUTES);
+  }
+
+  public static DatasetTriggerInfoRepo initAndGetInstance(int refreshFreqMin, Collection<String> dataSourceWhitelist) {
+    if (_instance == null) {
+      synchronized (DatasetTriggerInfoRepo.class) {
+        if (_instance == null) {
+          _instance = new DatasetTriggerInfoRepo(refreshFreqMin, dataSourceWhitelist);
+        }
+      }
+    }
+    return _instance;
+  }
+
+  public static DatasetTriggerInfoRepo getInstance() {
+    if (_instance == null)
+      throw new IllegalStateException("DatasetTriggerInfoRepo is not initialized before getInstance is called.");
+    return _instance;
+  }
+
+  public boolean isDatasetActive(String dataset) {
+    return datasetRefreshTimeMap.containsKey(dataset);
+  }
+
+  public long getLastUpdateTimestamp(String dataset) {
+    return (isDatasetActive(dataset)) ? datasetRefreshTimeMap.get(dataset) : 0;
+  }
+
+  public void setLastUpdateTimestamp(String dataset, long timestamp) {
+    datasetRefreshTimeMap.put(dataset, timestamp);
+  }
+
+  public void close() {
+    executorService.shutdown();
+    _instance = null;
+  }
+
+  private void refresh() {
+    Set<String> activeDatasetSet = new HashSet<>();
+    DetectionConfigManager detectionConfigDAO = DAO_REGISTRY.getDetectionConfigManager();
+    MetricConfigManager metricsConfigDAO = DAO_REGISTRY.getMetricConfigDAO();
+    DatasetConfigManager datasetConfigDAO = DAO_REGISTRY.getDatasetConfigDAO();
+    List<DetectionConfigDTO> detectionConfigDTOs = detectionConfigDAO.findAll();
```

Comment:
Can we add a method to get all active configs? I saw this pattern (getAll -> filter) multiple times.